src/components/DataProvider/index.tsx

```diff
@@ -0,0 +1,199 @@
+import * as React from 'react';
+import { UpdateDomains } from '../Griff';
+import {
+  calculateDomains,
+  isEqual,
+  highestPriorityDomain,
+} from '../../utils/domains';
+import { deleteUndefinedFromObject } from '../../utils/cleaner';
+import { placeholder } from '../../utils/placeholder';
+import { LoaderResult, Domain, LoaderReason } from '../../external';
+import { ScaledSeries, ScaledCollection } from '../../internal';
+import { debounce } from 'lodash';
+import { Props as ScalerProps } from '../Scaler';
+import Joiner from '../Joiner';
+import getItemsByItemId from '../../utils/itemsById';
+
+export interface Props extends ScalerProps {
+  series: ScaledSeries[];
+  collections: ScaledCollection[];
+
+  updateDomains: UpdateDomains;
+}
+
+interface State {
+  loaderResultsById: {
+    [seriesId: string]: LoaderResult & {
+      dataDomains: { time: Domain; x: Domain; y: Domain };
+    };
+  };
+}
+
+type SeriesById = { [seriesId: string]: ScaledSeries };
+
+type SeriesByCollectionId = { [seriesId: string]: ScaledSeries };
+
+interface RequestRecord {
+  timestamp: number;
+  timeSubDomain: Domain;
+}
+
+type FetchFunction = (
+  series: ScaledSeries,
+  reason: LoaderReason
+) => Promise<void>;
+
+const PLACEHOLDER_SUBDOMAIN = placeholder(0, 0);
+
+const DEFAULT_DATA_DOMAINS = {
+  time: PLACEHOLDER_SUBDOMAIN,
+  x: PLACEHOLDER_SUBDOMAIN,
+  y: PLACEHOLDER_SUBDOMAIN,
+};
+
+class DataProvider extends React.Component<Props, State> {
+  inFlightRequestsById: { [seriesId: string]: RequestRecord } = {};
+
+  fetchFunctionsById: { [seriesId: string]: FetchFunction } = {};
+
+  state: State = {
+    loaderResultsById: {},
+  };
+
+  componentDidUpdate() {
+    const { series } = this.props;
+
+    series.forEach((s: ScaledSeries) => {
+      const reason = this.getLoaderReason(s);
+      if (reason) {
+        let fetchFunction = this.fetchFunctionsById[s.id];
+        if (!fetchFunction) {
+          fetchFunction = debounce(this.fetchData(), 250, { leading: true });
+          this.fetchFunctionsById[s.id] = fetchFunction;
+        }
+        fetchFunction(s, reason);
+      }
+    });
+  }
+
+  getLoaderReason = (series: ScaledSeries) => {
```

Comment:
I don't see `INTERVAL` and `UPDATE_POINTS_PER_SERIES` have they been replaced or obsolete with a newer logic?